src/databases/block_database.cpp

```diff
@@ -275,6 +284,8 @@ bool block_database::update(const chain::block& block)
 
     BITCOIN_ASSERT(tx_start <= max_uint32);
     BITCOIN_ASSERT(tx_count <= max_uint16);
+    BITCOIN_ASSERT(!support_neutrino_filter_ ? true :
```

Comment:
A literal boolean value is never necessary in a ternary.

Reply:
Rewritten as logical or.